examples/fuzzing/rt_n12/fuzz.py

```diff
@@ -0,0 +1,173 @@
+    def my_nvram_get(self, ql:Qiling):
+        self.ql.log.info("NVRAM get call")
+        key_addr = self.ql.arch.regs.read("A0")
+        key = str(ql.mem.string(key_addr))
+        if self.dbg_level == QL_VERBOSE.DEBUG:
+            self.ql.log.info('key ' + key)
+        try:
+            val = self.nvram[key]
+        except Exception: # value not found, issue a blank one
+            val = "" # if this causes logic issues later on, edit the nvram.txt file
+        if self.dbg_level == QL_VERBOSE.DEBUG:
+            self.ql.log.info('value ' + val)
+
+        self.ql.mem.write(self.nvram_addr, bytes(val[::-1], 'utf-8'))
+        self.ql.arch.regs.write("V0", self.nvram_addr)
+        self.fake_return() # emulate return
+        return
+    def populate_nvram(self):
+        f = open('nvram', 'rb')
+        for line in f:
+            data = line.strip(b'\n')
+            pair = data.split(b'=')
+            key = str(pair[0].decode('utf-8')) # should not fail, unless NVRAM file is corrupt
+            if len(pair) != 2:
+                val = ""
+            else:
+                val = str(pair[1].decode('utf-8'))
+            self.nvram[key] = val
+
+    def my_nvram_set(self,ql:Qiling):
+        """
+        hook to emulate writing to NVRAM
+        """
+        value = self.ql.mem.string(ql.arch.regs.read("A1"))
+        key  = self.ql.mem.string(ql.arch.regs.read("A0"))
+       
+        self.nvram[value] = key
+        if self.dbg_level == QL_VERBOSE.DEBUG:
+            self.ql.log.info("Inside nvram set")
+            self.ql.log.info(value)
+            
+            self.ql.log.info(key)
+        self.fake_return()
+        return
+    def nvram_unset(self,ql:Qiling):
+        """
+        emulate clearing NVRAM
+        """
+        self.ql.log.info("fake unset")
+        self.fake_return()
+        return
+    def my_nvram_get_int(self, ql:Qiling):
+        """
+        hook emulating return an integer from NVRAM
+        """
+        self.ql.log.info("NVRAM get_int call")
+        key_addr = ql.arch.regs.read("A0")
+        key = str(ql.mem.string(key_addr))
+        val = self.nvram[key]
+        if val != '':
+            self.ql.arch.regs.write("V0", int(val, 16))
+        else:
+            self.ql.arch.regs.write("V0", 0x0)
+        self.fake_return()
+        return
+    def add_hooks(self):
+        """
+        hook all meaningful NVRAM calls
+        """
+        self.ql.hook_address(self.my_nvram_get, 0x00420690)
+        self.ql.hook_address(self.my_nvram_set, 0x004204b0)
+        self.ql.hook_address(self.my_nvram_get_int, 0x004201b0)
+        self.ql.hook_address(self.nvram_unset, 0x004206d0)
+    def __init__(self, dbg_level):
+        self.dbg_level = dbg_level
+        self.ql = Qiling(self.cmdline,rootfs=self.rootfs, verbose=dbg_level)
+        self.nvram = {}
+        self.nvram_addr = self.ql.mem.map_anywhere(size=4096, info='nvram label')
+        self.ql.fast_mode=True
```

Comment:
What is this?

Reply:
Added comment addressing that.

Reply:
I meant, what is `fast_mode`? It looks like something you added and never used.

Reply:
Removed.